pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -76,20 +84,58 @@ private void findSafeLiterals(AbstractApexNode<?> node) {
             int index = literal.jjtGetChildIndex();
             if (index == 0) {
                 if (node instanceof ASTVariableDeclaration) {
-                    addVariable(node);
-                } else {
+                    addVariable((ASTVariableDeclaration) node);
+                } else if (node instanceof ASTBinaryExpression) {
                     ASTVariableDeclaration parent = node.getFirstParentOfType(ASTVariableDeclaration.class);
-                    addVariable(parent);
+                    if (parent != null) {
+                        addVariable(parent);
+                    }
+                    ASTAssignmentExpression assignment = node.getFirstParentOfType(ASTAssignmentExpression.class);
+                    if (assignment != null) {
+                        ASTVariableExpression var = assignment.getFirstChildOfType(ASTVariableExpression.class);
+                        if (var != null) {
+                            addVariable(var);
+                        }
+                    }
 
                 }
             }
+        } else {
+            if (node instanceof ASTField) {
+                try {
+                    final Field f = node.getNode().getClass().getDeclaredField("fieldInfo");
```

Comment:
Similarly, in another PR I'm battling with ASTFieldDeclaration right now as there's no way to get the actual variable type out of it without using reflection. There are no child AST nodes that carry that data either.